admin/class-yoast-alerts.php

```diff
@@ -0,0 +1,245 @@
+			delete_user_meta( get_current_user_id(), $notification->get_dismissal_key() );
+
+			$this->output_ajax_response( $notification->get_type() );
+		}
+
+		die();
+	}
+
+	/**
+	 * Create AJAX response data
+	 *
+	 * @param string $type Alert type.
+	 */
+	private function output_ajax_response( $type ) {
+
+		$html = $this->get_view_html( $type );
+		echo json_encode(
+			array(
+				'html'  => $html,
+				'total' => count( self::$active_errors ) + count( self::$active_warnings ),
+			)
+		);
+	}
+
+	/**
+	 * Get the HTML to return in the AJAX request
+	 *
+	 * @param string $type Alert type.
+	 *
+	 * @return bool|string
+	 */
+	private function get_view_html( $type ) {
+
+		switch ( $type ) {
+			case 'error':
+			case 'warning':
+				$view = $type . 's';
+				break;
+			default:
+				return false;
+		}
+
+		// Re-collect alerts.
+		self::collect_alerts();
+
+		/** @noinspection PhpUnusedLocalVariableInspection */
+		$alerts_data = self::get_template_variables();
+
+		ob_start();
+		include WPSEO_PATH . 'admin/views/partial-alerts-' . $view . '.php';
+		$html = ob_get_clean();
+
+		return $html;
+	}
+
+	/**
+	 * Extract the Yoast Notification from the AJAX request
+	 *
+	 * @return null|Yoast_Notification
+	 */
+	private function get_notification_from_ajax_request() {
+
+		$notification_center = Yoast_Notification_Center::get();
+		$notification_id     = filter_input( INPUT_POST, 'notification' );
+
+		return $notification_center->get_notification_by_id( $notification_id );
+	}
+
+	/**
+	 * Show the alerts overview page
+	 */
+	public static function show_overview_page() {
+
+		/** @noinspection PhpUnusedLocalVariableInspection */
+		$alerts_data = self::get_template_variables();
+
+		include WPSEO_PATH . 'admin/views/alerts-dashboard.php';
+	}
+
+	/**
+	 * Collect the alerts and group them together
+	 */
+	public static function collect_alerts() {
+
+		$notification_center = Yoast_Notification_Center::get();
+
+		$notifications            = $notification_center->get_notifications();
+		self::$notification_count = count( $notifications );
+
+		self::$errors           = array_filter( $notifications, array( __CLASS__, 'filter_error_alerts' ) );
+		self::$dismissed_errors = array_filter( self::$errors, array( __CLASS__, 'filter_dismissed_alerts' ) );
+		self::$active_errors    = array_diff( self::$errors, self::$dismissed_errors );
+
+		self::$warnings           = array_filter( $notifications, array( __CLASS__, 'filter_warning_alerts' ) );
+		self::$dismissed_warnings = array_filter( self::$warnings, array( __CLASS__, 'filter_dismissed_alerts' ) );
+		self::$active_warnings    = array_diff( self::$warnings, self::$dismissed_warnings );
+	}
+
+	/**
+	 * Get the variables needed in the views
+	 *
+	 * @return array
+	 */
+	public static function get_template_variables() {
+
+		return array(
+			'metrics'  => array(
+				'total'    => self::$notification_count,
+				'active'   => count( self::$active_errors ) + count( self::$active_warnings ),
```

Comment:
I saw `count( self::$active_errors ) + count( self::$active_warnings )`  done before, maybe it worth consider extracting this to its own method.